### DISCUSSION TITLE
Mitsuba on musl / Alpine

### DISCUSSION THREAD

**Discussion by isolin** [mitsuba-renderer/mitsuba3#212] [Feature requests]
Hi! Would it please be possible to provide a python wheel for `musl` distros like Alpine Linux? That would be awesome since the corresponding docker containers would shrink quite a lot :) Currently `pip` tells there is no matching distribution of Mitsuba.

Here is a simple Dockerfile to demonstrate
```
FROM python:3.10-alpine
RUN pip install mitsuba
```

or just run

```sh
docker pull python:3.10-alpine
docker run -it python:3.10-alpine sh
# and inside the container
pip install mitsuba
```

**Comment by njroussel** [comment]
Hi @isolin !

My current understanding is that the NVidia drivers are linked against `glibc`, so if we were to provide a `musl` build of `mitsuba` it would be without any `cuda_*` variants :frowning_face: .
Also, if I recall correctly, there are a few places in our codebase where we make use of GNU extensions that are not in `musl`. Some code definitely would need to change. I haven't checked our external dependencies, but those might also rely on some specific `glibc` extensions.
So, it's not just a matter of building the wheel - we would need to rewrite some parts of the code.

I don't think this is a priority for us. But, if someone were to look into this and could make an assessment of how much code would need to change, I'd be happy to help.

**Comment by isolin** [comment]
Hi njroussel! I see, sounds like a lot of pain. In fact I would be fine with the `scalar`, but I guess a few Mb of the image are not worth the time if also other dependencies are specific to `glibc`. For now I took `FROM python:3.10-slim-bullseye` which makes 106Mb with Mitsuba. That is not bad at all :)
